cobalt/media/base/sbplayer_pipeline.cc

```diff
@@ -1093,9 +1093,10 @@ void SbPlayerPipeline::OnNeedData(DemuxerStream::Type type,
     // after the player has received enough audio for preroll, taking into
     // account that our estimate of playback time might be behind by
     // |kMediaTimeCheckInterval|.
-    if (timestamp_of_last_written_audio_ - seek_time_.ToSbTime() >
-        AdjustWriteDurationForPlaybackRate(audio_write_duration_for_preroll_,
-                                           playback_rate_)) {
+    if (!demuxer_->GetIsEndOfStreamReceived() &&
```

Comment:
The implementation here seems to ignore the audio write duration when at least one of the streams has received EOS from the web app, this may not be what we want:
Imagine the current playback position is 100s, and we have both audio and video buffered to 200s with MarkEndOfStream() called to signal that there won't be any further data for both streams.  Now the write duration is ignored, which shouldn't.

What we are trying to achieve is to ignore the audio write duration after the video EOS is written to the SbPlayer (or if the video EOS is received from the web app and close to be written to the SbPlayer).

Reply:
Ok, the new pr remove the limit after the video eos is written into the underlying player.